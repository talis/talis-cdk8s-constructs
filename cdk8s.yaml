language: typescript
imports:
  - k8s@1.30.0
