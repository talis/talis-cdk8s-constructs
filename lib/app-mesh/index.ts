export * from "./ingress-gateway-nlb";
export * from "./app-service";
