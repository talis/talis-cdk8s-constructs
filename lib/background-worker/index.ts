export * from "./background-worker-props";
export * from "./background-worker";
