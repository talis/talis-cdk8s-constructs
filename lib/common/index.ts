export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
