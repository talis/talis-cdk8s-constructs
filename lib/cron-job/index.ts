export * from "./cron-job-props";
export * from "./cron-job";
