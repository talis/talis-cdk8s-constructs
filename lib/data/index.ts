export * from "./config-map";
export * from "./secret";
export * from "./factory";
