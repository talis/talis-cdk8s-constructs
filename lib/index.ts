export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
