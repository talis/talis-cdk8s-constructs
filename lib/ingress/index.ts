export * from "./ingress-props";
export * from "./ingress";
