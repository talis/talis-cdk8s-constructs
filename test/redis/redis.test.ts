import { Chart, Testing } from "cdk8s";
import { Redis, RedisProps } from "../../lib";

const requiredProps = {
  release: "v1",
};

function synthRedis(props: RedisProps = requiredProps) {
  const chart = Testing.chart();
  new Redis(chart, "redis-test", props);
  const results = Testing.synth(chart);
  return results;
}

describe("Redis", () => {
  describe("Props", () => {
    test("Minimal required props", () => {
      const chart = Testing.chart();
      new Redis(chart, "redis-test", requiredProps);
      const results = Testing.synth(chart);
      expect(results).toMatchSnapshot();
    });

    test("All the props", () => {
      const app = Testing.app();
      const chart = new Chart(app, "test", {
        namespace: "test",
        labels: {
          app: "my-app",
          environment: "test",
          region: "local",
        },
      });
      const selectorLabels = {
        app: "my-app",
        role: "redis",
        instance: "test",
      };

      new Redis(chart, "redis-test", {
        ...requiredProps,
        selectorLabels,
      });
      const results = Testing.synth(chart);
      expect(results).toMatchSnapshot();
    });
  });

  describe("Container release", () => {
    test("Default container release", () => {
      const results = synthRedis();
      const job = results.find((obj) => obj.kind === "StatefulSet");
      expect(job).toHaveProperty(
        "spec.template.spec.containers[0].image",
        "redis:v1"
      );
    });

    test("Container release set explicitly", () => {
      const results = synthRedis({
        ...requiredProps,
        release: "12345",
      });
      const job = results.find((obj) => obj.kind === "StatefulSet");
      expect(job).toHaveProperty(
        "spec.template.spec.containers[0].image",
        "redis:12345"
      );
    });
  });
});
