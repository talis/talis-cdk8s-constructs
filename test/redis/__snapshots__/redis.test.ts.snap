// Vitest Snapshot v1

exports[`Redis > Props > All the props 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "app": "my-app",
        "environment": "test",
        "instance": "test",
        "region": "local",
        "release": "v1",
        "role": "redis",
      },
      "name": "test-redis-test-c8b56c58",
      "namespace": "test",
    },
    "spec": {
      "clusterIP": "None",
      "ports": [
        {
          "port": 6379,
          "protocol": "TCP",
        },
      ],
      "selector": {
        "app": "my-app",
        "instance": "test",
        "role": "redis",
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": {
      "labels": {
        "app": "my-app",
        "environment": "test",
        "instance": "test",
        "region": "local",
        "release": "v1",
        "role": "redis",
      },
      "name": "test-redis-test-redis-test-sts-c82d6e34",
      "namespace": "test",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "app": "my-app",
          "instance": "test",
          "role": "redis",
        },
      },
      "serviceName": "test-redis-test-c8b56c58",
      "template": {
        "metadata": {
          "labels": {
            "app": "my-app",
            "environment": "test",
            "instance": "test",
            "region": "local",
            "release": "v1",
            "role": "redis",
          },
        },
        "spec": {
          "containers": [
            {
              "command": [
                "redis-server",
                "--appendonly",
                "yes",
              ],
              "env": [
                {
                  "name": "MASTER",
                  "value": "true",
                },
              ],
              "image": "redis:v1",
              "livenessProbe": {
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "tcpSocket": {
                  "port": 6379,
                },
                "timeoutSeconds": 5,
              },
              "name": "redis",
              "ports": [
                {
                  "containerPort": 6379,
                },
              ],
              "readinessProbe": {
                "exec": {
                  "command": [
                    "redis-cli",
                    "ping",
                  ],
                },
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "timeoutSeconds": 5,
              },
              "resources": {
                "limits": {
                  "cpu": "100m",
                  "memory": "250Mi",
                },
              },
              "volumeMounts": [
                {
                  "mountPath": "/redis-master-data",
                  "name": "data",
                },
              ],
            },
          ],
          "priorityClassName": "test",
          "volumes": [
            {
              "emptyDir": {},
              "name": "data",
            },
          ],
        },
      },
    },
  },
]
`;

exports[`Redis > Props > Minimal required props 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "instance": "redis-test",
        "release": "v1",
        "role": "redis",
      },
      "name": "test-redis-test-c8b56c58",
    },
    "spec": {
      "clusterIP": "None",
      "ports": [
        {
          "port": 6379,
          "protocol": "TCP",
        },
      ],
      "selector": {
        "instance": "redis-test",
        "role": "redis",
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": {
      "labels": {
        "instance": "redis-test",
        "release": "v1",
        "role": "redis",
      },
      "name": "test-redis-test-redis-test-sts-c82d6e34",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "instance": "redis-test",
          "role": "redis",
        },
      },
      "serviceName": "test-redis-test-c8b56c58",
      "template": {
        "metadata": {
          "labels": {
            "instance": "redis-test",
            "release": "v1",
            "role": "redis",
          },
        },
        "spec": {
          "containers": [
            {
              "command": [
                "redis-server",
                "--appendonly",
                "yes",
              ],
              "env": [
                {
                  "name": "MASTER",
                  "value": "true",
                },
              ],
              "image": "redis:v1",
              "livenessProbe": {
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "tcpSocket": {
                  "port": 6379,
                },
                "timeoutSeconds": 5,
              },
              "name": "redis",
              "ports": [
                {
                  "containerPort": 6379,
                },
              ],
              "readinessProbe": {
                "exec": {
                  "command": [
                    "redis-cli",
                    "ping",
                  ],
                },
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "timeoutSeconds": 5,
              },
              "resources": {
                "limits": {
                  "cpu": "100m",
                  "memory": "250Mi",
                },
              },
              "volumeMounts": [
                {
                  "mountPath": "/redis-master-data",
                  "name": "data",
                },
              ],
            },
          ],
          "priorityClassName": "database",
          "volumes": [
            {
              "emptyDir": {},
              "name": "data",
            },
          ],
        },
      },
    },
  },
]
`;
